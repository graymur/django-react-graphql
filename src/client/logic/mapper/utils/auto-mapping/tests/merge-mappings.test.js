const mergeMappings = require('../merge-mappings').default

test('merges mappings', () => {
	const mappings = [
		[
			{
				from: 'from_firstName',
				to: 'to_firstName',
			},
			{
				from: 'from_lastName',
				to: 'to_lastName',
			},
			{
				from: 'from_email',
				to: 'to_email',
			},
		],
		[
			{
				from: 'from_firstName', // should not be in final mapping
				to: 'to_firstName2',
			},
			{
				from: 'from_middleName',
				to: 'to_middleName',
			},
		],
		[
			{
				from: 'from_firstName', // should not be in final mapping
				to: 'to_firstName3',
			},
			{
				from: 'from_middleName', // should not be in final mapping
				to: 'to_middleName2',
			},
			{
				from: 'from_phone',
				to: 'to_phone',
			},
		],
	]

	const result = mergeMappings(mappings)
	expect(result).toMatchSnapshot()
})
