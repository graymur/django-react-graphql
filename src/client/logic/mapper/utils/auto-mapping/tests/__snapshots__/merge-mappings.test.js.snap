// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merges mappings 1`] = `
Array [
  Object {
    "from": "from_firstName",
    "to": "to_firstName",
  },
  Object {
    "from": "from_lastName",
    "to": "to_lastName",
  },
  Object {
    "from": "from_email",
    "to": "to_email",
  },
  Object {
    "from": "from_middleName",
    "to": "to_middleName",
  },
  Object {
    "from": "from_phone",
    "to": "to_phone",
  },
]
`;
